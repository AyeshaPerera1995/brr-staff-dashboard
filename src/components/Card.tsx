import React from "react";

type CardProps = {
  title: string;
  value: string | number;
  icon?: React.ReactNode;
};

export default function Card({ title, value, icon }: CardProps) {
  return (
    <div className="bg-white hover:shadow-lg transition-shadow duration-300 p-6 rounded-2xl flex flex-col gap-2 w-full border border-gray-200">
      <div className="flex items-center justify-between text-gray-600">
        <h3 className="text-md font-semibold">{title}</h3>
        {icon && <div data-testid="card-icon" className="text-2xl text-indigo-500 cursor-pointer">{icon}</div>}
      </div>
      <div className="text-3xl font-bold text-gray-900">{value}</div>
    </div>
  );
}

