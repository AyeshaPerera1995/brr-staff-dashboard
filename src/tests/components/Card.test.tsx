import React from 'react'; 
import { render, screen } from '@testing-library/react';
import Card from '@/components/Card';
import { FiAlertCircle } from 'react-icons/fi';

describe("Card Component", () => {

  it("renders the title and value correctly", () => {
    render(<Card title="Open Tickets" value={5} />);
    expect(screen.getByText("Open Tickets")).toBeInTheDocument();
    expect(screen.getByText("5")).toBeInTheDocument();
  });

  it("renders value correctly when it is a string", () => {
    render(<Card title="Status" value="Active" />);
    expect(screen.getByText("Active")).toBeInTheDocument();
  });

  it("renders icon if provided", () => {
    render(<Card title="Open Tickets" value={5} icon={<FiAlertCircle />} />);
    const icon = screen.getByTestId("card-icon");
    expect(icon).toBeInTheDocument();
  });

  it("does not render icon if not provided", () => {
    render(<Card title="Open Tickets" value={5} />);
    expect(screen.queryByTestId("card-icon")).toBeNull();
  });

  it("matches snapshot", () => {
    const { container } = render(<Card title="Open Tickets" value={5} icon={<FiAlertCircle />} />);
    expect(container).toMatchSnapshot();
  });
});