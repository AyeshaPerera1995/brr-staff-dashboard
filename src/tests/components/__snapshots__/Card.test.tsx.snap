// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
<div>
  <div
    class="bg-white hover:shadow-lg transition-shadow duration-300 p-6 rounded-2xl flex flex-col gap-2 w-full border border-gray-200"
  >
    <div
      class="flex items-center justify-between text-gray-600"
    >
      <h3
        class="text-md font-semibold"
      >
        Open Tickets
      </h3>
      <div
        class="text-2xl text-indigo-500 cursor-pointer"
        data-testid="card-icon"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="8"
            y2="12"
          />
          <line
            x1="12"
            x2="12.01"
            y1="16"
            y2="16"
          />
        </svg>
      </div>
    </div>
    <div
      class="text-3xl font-bold text-gray-900"
    >
      5
    </div>
  </div>
</div>
`;
